### PR TITLE
Supply a reason when removing an item from the mempool

### PR DESCRIPTION
**Pull request by aqk**
No user visible changes. This change will let the mempool know why a caller is removing a SpendBundle.

This is a required part of the Fee Estimator backend

### CODE REVIEW

File: chia/full_node/mempool.py [hunk 1]
```diff
@@ -17,6 +18,18 @@
 from chia.util.ints import uint64
 
 
+class MempoolRemoveReason(IntEnum):
```
Comment (arvidn) [inline chia/full_node/mempool.py:21]: this looks just like an enum. does this have any benefits over an enum?

File: chia/full_node/mempool.py [hunk 2]
```diff
@@ -73,7 +86,8 @@ def remove_from_pool(self, items: List[bytes32]) -> None:
             self.total_mempool_cost -= item.cost
             assert self.total_mempool_cost >= 0
             mempool_info = self.get_mempool_info()
-            self.fee_estimator.remove_mempool_item(mempool_info, item)
+            if reason != MempoolRemoveReason.BLOCK_INCLUSION:
```
Comment (arvidn) [inline chia/full_node/mempool.py:89]: this looks like a change in behavior. would you mind explaining it?

File: chia/full_node/mempool.py (hunk 1 above)
Comment (aqk) [reply chia/full_node/mempool.py:21]: Nope! I just got in the habit of using IntEnum from us using Streamable to manage on-the-wire values. I'll change that to Enum.

File: chia/full_node/mempool.py (hunk 2 above)
Comment (aqk) [reply chia/full_node/mempool.py:89]: Yes. This is the first in a series of small fixes that fix broken behaviour in the fee estimator, specifically how and where the code is called from.

This particular line fixes the issue that MempoolItems were being marked as removed in the fee estimator even though the reason they were being removed was that they had successfully gotten into a block.

Comment (aqk) [reply chia/full_node/mempool.py:83]: I could take that line out and put it in the PR that does have a user-visible fix.

Let me know what you think.

Comment (aqk) [reply chia/full_node/mempool.py:83]: Yeah, I think I'll do that.

**Review by arvidn — DISMISSED** [review]

**Review by almogdepaz — APPROVED** [review]

**Pull request closed without merging**